vizro-core/tests/e2e/test_component_library.py

```diff
@@ -0,0 +1,88 @@
+import dash_bootstrap_components as dbc
+import pandas as pd
+from dash import Dash, html
+from e2e_asserts import assert_image_equal
+
+from vizro.figures.library import kpi_card, kpi_card_reference
+
+df_kpi = pd.DataFrame(
+    {
+        "Actual": [100, 200, 700],
+        "Reference": [100, 300, 500],
+        "Category": ["A", "B", "C"],
+    }
+)
+
+example_cards = [
+    kpi_card(data_frame=df_kpi, value_column="Actual", title="KPI with value"),
+    kpi_card(
+        data_frame=df_kpi,
+        value_column="Actual",
+        title="KPI with aggregation",
+        agg_func="median",
+    ),
+    kpi_card(
+        data_frame=df_kpi,
+        value_column="Actual",
+        title="KPI formatted",
+        value_format="${value:.2f}",
+    ),
+    kpi_card(
+        data_frame=df_kpi,
+        value_column="Actual",
+        title="KPI with icon",
+        icon="shopping_cart",
+    ),
+]
+
+example_reference_cards = [
+    kpi_card_reference(
+        data_frame=df_kpi,
+        value_column="Actual",
+        reference_column="Reference",
+        title="KPI ref. (pos)",
+    ),
+    kpi_card_reference(
+        data_frame=df_kpi,
+        value_column="Actual",
+        reference_column="Reference",
+        agg_func="median",
+        title="KPI ref. (neg)",
+    ),
+    kpi_card_reference(
+        data_frame=df_kpi,
+        value_column="Actual",
+        reference_column="Reference",
+        title="KPI ref. formatted",
+        value_format="{value}€",
+        reference_format="{delta}€ vs. last year ({reference}€)",
+    ),
+    kpi_card_reference(
+        data_frame=df_kpi,
+        value_column="Actual",
+        reference_column="Reference",
+        title="KPI ref. with icon",
+        icon="shopping_cart",
+    ),
+]
+
+
+def test_kpi_card_component_library(dash_duo, get_test_name):
+    app = Dash(__name__, external_stylesheets=[dbc.themes.BOOTSTRAP])
+    app.layout = dbc.Container(
+        [
+            html.H1(children="KPI Cards"),
+            dbc.Stack(
+                children=[
+                    dbc.Row([dbc.Col(kpi_card) for kpi_card in example_cards]),
+                    dbc.Row([dbc.Col(kpi_card) for kpi_card in example_reference_cards]),
+                ],
+                gap=4,
+            ),
+        ]
+    )
+    dash_duo.start_server(app)
+    dash_duo.wait_for_page(timeout=20)
+    dash_duo.wait_for_element("div[class='card-kpi card']")
+    assert_image_equal(dash_duo.driver, get_test_name)
```

Comment:
I know it was my idea to refactor it like this... But I didn't realise exactly what it would look like until you did it, and now it looks a bit weird 😅 `assert_image_equal` really sounds like it should be taking in images rather than `browserdriver` and `testname`.

I suggest breaking this down like this:
```
# remove get_test_name fixture and just use request directly given it's only used once
# this code might be repeated a lot across tests but that's ok, I think it makes sense
result_image_path, expected_image_path  = make_screenshot_paths(request.node.name)
dash_duo.save_screenshot(result_image_path)
assert_image_equal(result_image_path, expected_image_path)
```

```
# e2e_asserts.py
def make_screenshot_paths(test_name):
    # do check this, I might have done it wrong...
    result_image_path = f"{test_image_name}_branch.png"
    expected_image_path = f"tests/e2e/screenshots/{test_image_name.replace('test', 'base')}.png"
    return  result_image_path, expected_image_path

def assert_image_equal(result_image_path, expected_image_path):
    """Comparison logic and diff files creation."""
    result_image = cv2.imread(result_image_path)    
    expected_image = cv2.imread(expected_image_path)
    
    try:
        # basically the same as now with some renaming of variables
```

Reply:
Did it a bit different, but should be ok